### PR TITLE
6.3 hardening 5

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java [hunk 1]
```diff
@@ -421,7 +421,9 @@
 
   /**
    * @since 4.0
+   * @deprecated no more used since 6.3. See https://jira.sonarsource.com/browse/SONAR-8610
    */
+  @Deprecated
   String HOURS_IN_DAY = "sonar.technicalDebt.hoursInDay";
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/utils/Durations.java [hunk 2]
```diff
@@ -62,76 +60,84 @@ public Duration create(long minutes) {
...
   /**
    * Return the formatted work duration.
    * <br>
-   * Example : format(Locale.FRENCH, Duration.encode("9d 10h"), DurationFormat.SHORT) -&gt; 10j 2h (if sonar.technicalDebt.hoursInDay property is set to 8)
+   * Example : format(Locale.FRENCH, Duration.encode("9d 10h"), DurationFormat.SHORT) -&gt; 10d 2d
    *
+   * @deprecated since 6.3 as the {@link Locale#ENGLISH} is always used. Use {@link #format(Duration)} instead
    */
+  @Deprecated
   public String format(Locale locale, Duration duration) {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/utils/Durations.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/utils/Durations.java [hunk 3]
```diff
@@ -62,76 +60,84 @@ public Duration create(long minutes) {
...
 
   /**
    * Return the formatted work duration.
-   * <br>
-   * Example : format(Locale.FRENCH, Duration.encode("9d 10h"), DurationFormat.SHORT) -&gt; 10j 2h (if sonar.technicalDebt.hoursInDay property is set to 8)
    *
+   * @deprecated since 6.3 as the {@link Locale#ENGLISH} is always used. Use {@link #format(Duration)} instead
    */
+  @Deprecated
   public String format(Locale locale, Duration duration, DurationFormat format) {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/utils/Durations.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/utils/Durations.java [hunk 4]
```diff
@@ -36,6 +32,16 @@
...
+  private static final String HOURS_FORMAT = "%sh";
+  private static final String DAYS_FORMAT = "%sd";
+
+  private static final int HOURS_IN_DAY = 8;
+
+  /**
+   * @deprecated since 6.3, only one format is available
+   */
+  @Deprecated
   public enum DurationFormat {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/utils/Durations.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues
* ![INFO][INFO] 4 info

Watch the comments in this conversation to review them.

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'

**Comment by SonarTech** [comment]
SonarQube analysis reported 10 issues
* ![MAJOR][MAJOR] 5 major
* ![INFO][INFO] 5 info

Watch the comments in this conversation to review them.

#### 6 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BulkRuleActivationActions.java#L46](https://github.com/SonarSource/sonarqube/blob/57c4dff128c5cd90c196ef24988856e6581fbf9c/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/BulkRuleActivationActions.java#L46): Remove this unused "i18n" private field. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [BulkRuleActivationActions.java#L109](https://github.com/SonarSource/sonarqube/blob/57c4dff128c5cd90c196ef24988856e6581fbf9c/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/BulkRuleActivationActions.java#L109): Make "writeResponse" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)
1. ![MAJOR][MAJOR] [WebServiceEngine.java#L64](https://github.com/SonarSource/sonarqube/blob/57c4dff128c5cd90c196ef24988856e6581fbf9c/server/sonar-server/src/main/java/org/sonar/server/ws/WebServiceEngine.java#L64): Remove this unused "i18n" private field. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [WebServiceEngine.java#L107](https://github.com/SonarSource/sonarqube/blob/57c4dff128c5cd90c196ef24988856e6581fbf9c/server/sonar-server/src/main/java/org/sonar/server/ws/WebServiceEngine.java#L107): NullPointerException might be thrown as 'action' is nullable here [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259)
1. ![MAJOR][MAJOR] [WebServiceEngine.java#L136](https://github.com/SonarSource/sonarqube/blob/57c4dff128c5cd90c196ef24988856e6581fbf9c/server/sonar-server/src/main/java/org/sonar/server/ws/WebServiceEngine.java#L136): Make "sendErrors" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)
1. ![INFO][INFO] [CoreProperties.java#L192](https://github.com/SonarSource/sonarqube/blob/57c4dff128c5cd90c196ef24988856e6581fbf9c/sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java#L192): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 1 major
* ![INFO][INFO] 4 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WebServiceEngine.java#L133](https://github.com/SonarSource/sonarqube/blob/e9e079da4601649d20972302d6b266771fd9cf6b/server/sonar-server/src/main/java/org/sonar/server/ws/WebServiceEngine.java#L133): Make "sendErrors" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: it/it-tests/src/test/java/it/qualityModel/DebtConfigurationRule.java [hunk 5]
```diff
@@ -33,7 +33,6 @@
  */
 public class DebtConfigurationRule extends ExternalResource {
 
-  private static final String HOURS_IN_DAY_PROPERTY = "sonar.technicalDebt.hoursInDay";
```
Comment (julienlancelot) [inline it/it-tests/src/test/java/it/qualityModel/DebtConfigurationRule.java]: No, but I'll fix it if it's the case

File: sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java [hunk 6]
```diff
@@ -428,7 +428,9 @@
...
```
Comment (julienlancelot) [inline sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java]: Indeed !

**Comment by julienlancelot** [comment]
@teryk Please review my last commit

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 1 major
* ![INFO][INFO] 4 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WebServiceEngine.java#L133](https://github.com/SonarSource/sonarqube/blob/0431cefd2edf006954bfd3d18ba1f842085e1b9f/server/sonar-server/src/main/java/org/sonar/server/ws/WebServiceEngine.java#L133): Make "sendErrors" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 1 major
* ![INFO][INFO] 4 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WebServiceEngine.java#L133](https://github.com/SonarSource/sonarqube/blob/57561f65d0e8189a9ab0732b53057e0df46ec5ab/server/sonar-server/src/main/java/org/sonar/server/ws/WebServiceEngine.java#L133): Make "sendErrors" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Pull request merged**